hack/update-istio-samples.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+# Base directory containing the sample application folders
+SAMPLES_DIR="tests/e2e/samples"
+# Quay repository where images will be copied
+QUAY_HUB="quay.io/sail-dev"
+
+# Array to store unique images that need to be copied
+declare -A IMAGES_TO_COPY
+
+# Iterate over all subdirectories in the samples directory
+for dir in "$SAMPLES_DIR"/*/; do
+    # Skip if it's not a directory
+    [[ ! -d "$dir" ]] && continue
+
+    name=$(basename "$dir")
+    kustomization_file="${dir}kustomization.yaml"
```

Comment:
```suggestion
    kustomization_file="${dir}/kustomization.yaml"
```

I know it's probably not necessary now, but it will avoid future errors should someone change how the script searches for directories